### PR TITLE
Break game and protections

### PR DESCRIPTION
**Pull request by maybegreat48**
More coming soon

Partially alleviates #3195, but it can never be fixed completely

### CODE REVIEW

**Comment by gir489returns** [comment]
I am OK with protecting against this, however, I am not OK with being able to attack people with this. Does this crash their game? If so, this should be removed.

**Comment by maybegreat48** [comment]
> Does this crash their game? If so, this should be removed.

Nope, it works like start activity and things like that. The player can leave at any time, this doesn't softlock their game

There are variations of this exploit that can be used to crash people and even be used to manipulate certain globals

**Comment by Rxann** [comment]
> I am OK with protecting against this, however, I am not OK with being able to attack people with this. Does this crash their game? If so, this should be removed.

No it doesn't. It just leaves them in a paralyzed state of sorts.You still have the ability to leave.

**Comment by lonelybud** [comment]
will it destroy public lobby missions and gameplay? Protections should be fine.

Edit: i also hate yimmenu sometimes bcz it has those features called start activity's, teleport everyone, and what is missing are crashes and those can also be freely provided via lua.

**Comment by maybegreat48** [comment]
> will it destroy public lobby missions and gameplay? Protections should be fine

Well yes, it's essentially a kick but the player stays on your machine. It won't break missions silently or for other people so there's that

> Edit: i also hate yimmenu bcz it has those features called start activity's, teleport everyone, and what is missing are crashes and those can also be freely provided via lua.

Anything can be abused if you're smart enough. I've seen spammers promoting their porn website ("erotic cinema") host their sessions with the multiplexer (from YimMenu) to get people to join their session. They also increase the player limit from 30 to 32 (also from YimMenu)

**Comment by gir489returns** [comment]
> Anything can be abused if you're smart enough. I've seen spammers promoting their porn website ("erotic cinema") host their sessions with the multiplexer (from YimMenu) to get people to join their session. They also increase the player limit from 30 to 32 (also from YimMenu)

That's awfully egotistical and assumptive to presume they were using YimMenu, or even that they stole it from us. While it's highly likely they are and probably did, we're not the only ones with Visual C++ and a copy of IDA...

**Comment by Rxann** [comment]
> > Anything can be abused if you're smart enough. I've seen spammers promoting their porn website ("erotic cinema") host their sessions with the multiplexer (from YimMenu) to get people to join their session. They also increase the player limit from 30 to 32 (also from YimMenu)
> 
> 
> 
> That's awfully egotistical and assumptive to presume they were using YimMenu, or even that they stole it from us. While it's highly likely they are and probably did, we're not the only ones with Visual C++ and a copy of IDA...

Everyone steals from YimMenu. Stand, I have heard of R* games themselves even doing it. And, I imagine there is a boat load of shitty menus that steal almost every feature.

Edit:
We also just released these features so the biggest culprits haven't likely had time to put them in their own menus yet.

**Comment by lonelybud** [comment]
I m not the only one to decide what is good or bad for menu but at least we should remove "all" related commands which execute on every player. Example - `breakgameall`. I mean why break everyone's game?

**Comment by lonelybud** [comment]
> Anything can be abused if you're smart enough. I've seen spammers promoting their porn website ("erotic cinema") host their sessions with the multiplexer (from YimMenu) to get people to join their session. They also increase the player limit from 30 to 32 (also from YimMenu)

well they are doing that. Creating solo sessions and are hosting lobbies as i said before 95% of chinese sessions i have joined automatically are hosted by spam bots and this has happened recently after new changes added in yimmenu

**Comment by gir489returns** [comment]
> Everyone steals from YimMenu. Stand, I have heard of R* games themselves even doing it. And, I imagine there is a boat load of shitty menus that steal almost every feature.

Literally my uncle dad works at Nintendo style speculation asserted as facts. Retarded.

**Comment by Rxann** [comment]
> I m not the only one to decide what is good or bad for menu but at least we should remove "all" related commands which execute on every player. Example - `breakgameall`. I mean why break everyone's game?

Once Stand inevitably gets their grubby hands on this feature, I can guarantee they will have a break all option. We might as well have it because every other P2C will. At the end of the day, we are a mod menu.

**Comment by maybegreat48** [comment]
> Literally my uncle dad works at Nintendo style speculation asserted as facts. Retarded.

It isn't always true but it is most of the time. Just look at commit dates over the past few years and observe what P2Cs add soon after in their changelogs. This isn't necessarily a bad thing, of course; we would definitely try to add an interesting feature if others have it

> I m not the only one to decide what is good or bad for menu but at least we should remove "all" related commands which execute on every player. Example - breakgameall. I mean why break everyone's game?

This harkens backs to the time when toxic features were not allowed in YimMenu. We eventually decided to add them since many people would straight up not use a menu that do not have these options

> Once Stand inevitably gets their grubby hands on this feature, I can guarantee they will have a break all option. We might as well have it because every other P2C will. At the end of the day, we are a mod menu.

Stand isn't even the worst of it. There are scammers that re-sell YimMenu with little to no modifications

**Comment by maybegreat48** [comment]
I've removed the `breakgameall` button. You can still use it in the command executor, but the most of the (s)kids can't

File: src/hooks/protections/receive_net_message.cpp [hunk 1]
```diff
@@ -255,6 +264,24 @@ namespace big
...
+				{
+					LOG(WARNING) << "We're being replaced";
+					return true;
+				}
+
+				if (bubble != 0)
+				{
+					LOG(WARNING) << "Wrong bubble: " << bubble;
+				}
+			}
```
Comment (lonelybud) [inline src/hooks/protections/receive_net_message.cpp:285]: add `break;` statement

**Review by Yimura — APPROVED** [review]

**Pull request merged**